src/explore-education-statistics-admin/src/pages/release/edit-release/content/ReleaseContext.tsx

```diff
@@ -0,0 +1,237 @@
+          });
+        }
+      });
+    }
+    case 'ADD_BLOCK_TO_SECTION': {
+      return produce<State>(state, draft => {
+        const { block, meta } = action.payload;
+        const { sectionId, sectionKey } = meta;
+        if (
+          draft.release === undefined ||
+          typeof draft.release[sectionKey] !== 'object'
+        ) {
+          throw new Error('ADD_BLOCK_TO_SECTION: failed');
+        } else if (sectionKey === 'content') {
+          draft.release[sectionKey] = draft.release[sectionKey].map(section => {
+            if (section.id === sectionId) {
+              return {
+                ...section,
+                content: section.content
+                  ? [
+                      ...section.content,
+                      { ...block, comments: block.comments || [] },
+                    ]
+                  : [{ ...block, comments: block.comments || [] }],
+              } as ContentSection<EditableContentBlock>;
+            }
+            return section;
+          });
+        } else if (
+          (draft.release[sectionKey] as ContentSection<EditableContentBlock>)
+            .content
+        ) {
+          ((draft.release[sectionKey] as ContentSection<EditableContentBlock>)
+            .content as EditableContentBlock[]).push(block);
+        } else {
+          (draft.release[sectionKey] as ContentSection<
+            EditableContentBlock
+          >).content = [block];
+        }
+      });
+    }
+    case 'UPDATE_SECTION_CONTENT': {
+      return produce<State>(state, draft => {
+        const { sectionContent, meta } = action.payload;
+        const { sectionId, sectionKey } = meta;
+        if (
+          draft.release === undefined ||
+          typeof draft.release[sectionKey] !== 'object'
+        ) {
+          throw new Error('ADD_BLOCK_TO_SECTION: failed');
+        } else if (sectionKey === 'content') {
+          draft.release[sectionKey] = draft.release[sectionKey].map(section => {
+            if (section.id === sectionId) {
+              return { ...section, content: sectionContent };
+            }
+            return section;
+          });
+        } else {
+          (draft.release[sectionKey] as ContentSection<
+            EditableContentBlock
+          >).content = sectionContent;
+        }
+      });
+    }
+    case 'ADD_CONTENT_SECTION': {
+      return produce<State>(state, draft => {
+        const { section } = action.payload;
+        if (draft.release) draft.release.content.push(section);
+      });
+    }
+    case 'SET_CONTENT': {
+      return produce<State>(state, draft => {
+        const { content } = action.payload;
+        if (draft.release) draft.release.content = content;
+      });
+    }
+    case 'UPDATE_CONTENT_SECTION': {
+      return produce<State>(state, draft => {
+        const { section, meta } = action.payload;
+        const { sectionId } = meta;
+        if (draft.release)
+          draft.release.content = draft.release.content.map(
```

Comment:
Take the suggestions from above to write the rest of this branch in a more Immer idiomatic way.